presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -67,6 +67,21 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
         ImmutableList.Builder<ByteCodeNode> defaultBucket = ImmutableList.builder();
         ImmutableSet.Builder<Object> constantValuesBuilder = ImmutableSet.builder();
 
+        boolean areSmallPrimitives = false;
+        if (values.size() < 1000 && javaType == long.class) {
+            areSmallPrimitives = true;
+            for (RowExpression testValue : values) {
+                if (testValue instanceof ConstantExpression && ((ConstantExpression) testValue).getValue() != null) {
```

Comment:
@electrum  The logic is correct. For non constants, we are going to test them in the default branch anyways. Therefore, they shouldn't affect the decision on whether lookupswitch on hash value or lookupswitch on original value will be generated.